### PR TITLE
Update documentation about AutocompleteInput and AutocompleteArrayInput

### PR DESCRIPTION
**Pull request by djhi**

### CODE REVIEW

**Comment by djhi** [comment]
Fixes #8545

File: docs/AutocompleteArrayInput.md [hunk 1]
```diff
@@ -337,25 +337,37 @@ const choices = [
...
 
 <AutocompleteArrayInput source="authors" choices={choices} optionText={optionRenderer} />
 ```
 
-`optionText` also accepts a React Element, that will be rendered inside a [`<RecordContext>`](./useRecordContext.md) using the related choice as the `record` prop. You can use Field components there.
+**Tip**: Make sure you provide a stable reference to the function passed as `optionText`. Either declare it outside the component render function or wrap it inside a [`useCallback`](https://react.dev/reference/react/useCallback).
+
+`optionText` also accepts a React Element, that will be rendered inside a [`<RecordContext>`](./useRecordContext.md) using the related choice as the `record` prop. You can use Field components there. However, using an element as `optionText` implies that you also set two more props, `inputText` and `matchSuggestion`. See [Using A Custom Element For Options](#using-a-custom-element-for-options) for more details.
+
+`optionText` is also useful when the choices are records [fetched from another resource](#fetching-choices), and `<AutocompleteArrayInput>` is a child of a [`<ReferenceArrayInput>`](./ReferenceArrayInput.md).
```
Comment (slax57) [inline docs/AutocompleteArrayInput.md:351]: To me, this line is redundant with line 325

File: docs/AutocompleteArrayInput.md [hunk 2]
```diff
@@ -337,25 +337,37 @@ const choices = [
...
+
+But if you set the `optionText` prop, react-admin uses it instead of relying on `recordRepresentation`.
+
+```jsx
+import { AutocompleteArrayInput, ReferenceArrayInput } from 'react-admin';
 
-<AutocompleteArrayInput source="authors" choices={choices} optionText={<FullNameField />}/>
+<ReferenceArrayInput label="Author" source="author_id" reference="authors">
+    <AutocompleteArrayInput optionText="last_name" />
+</ReferenceArrayInput>
```
Comment (slax57) [inline docs/AutocompleteArrayInput.md:370]: This example is redundant with example given lines 328-330.

Besides, you removed the example where we actually pass a React element to `optionText`. We should keep at least one such example IMO.

**Review by slax57 — CHANGES_REQUESTED** [review]
Also, the title of the PR is misleading, because you also changed something in the code, not only the docs

**Review by slax57 — APPROVED** [review]

**Pull request merged**